Octokit/Http/Credentials.cs

```diff
@@ -13,23 +13,23 @@ private Credentials()
             AuthenticationType = AuthenticationType.Anonymous;
         }
 
-        public Credentials(string token)
-        {
-            Ensure.ArgumentNotNullOrEmptyString(token, "token");
-
-            Login = null;
-            Password = token;
-            AuthenticationType = AuthenticationType.Oauth;
-        }
-
-        public Credentials(string login, string password)
+        public Credentials(string login, string password, AuthenticationType authenticationType = AuthenticationType.Basic)
```

Comment:
again, to avoid a breaking change (even a parameter with a default value can still be a breaking change binarily speaking) you should add new `ctor`/s for these changes

Reply:
Are you sure? AFAIK the run time takes care about it...

Reply:
See this link, for example
https://haacked.com/archive/2010/08/10/versioning-issues-with-optional-arguments.aspx/